libraries/chain/include/graphene/chain/vesting_balance_object.hpp

```diff
@@ -118,10 +119,44 @@ namespace graphene { namespace chain {
       void on_withdraw(const vesting_policy_context& ctx);
    };
 
+   /**
+    * @brief Cant withdraw anything while dormant mode is true, linear policy after that changes.
+    *
+    * This vesting balance type is needed to register SON users where balance may be claimable only after
+    * the SON object is deleted(plus a linear policy).
+    * When deleting a SON member the dormant_mode will change and the linear policy will became active.
+    *
+    * @note New funds may not be added to a dormant vesting balance.
+    */
+      struct dormant_vesting_policy
```

Comment:
Can't we use dormant template class, so that in future if we have to apply dormant functionality to existing or new policies we can reuse the same template class?

Reply:
I think you are right, i misunderstood the document. working on a fix for this. will update. thx.

Reply:
Please check https://github.com/peerplays-network/peerplays/pull/183#issuecomment-543478319

Reply:
I think this approach also looks fine.
I feel as long as we don't duplicate the code and abstract the dormant policy it is fine.
The code looks fine to me. Thanks.